lib/cmds/ocr_cmds/get-suggestions.js

```diff
@@ -0,0 +1,191 @@
+  if (!response || !response.length) return;
+  const filteredSuggestions = [];
+  const potentialSuggestions = response.map(_ => ({ suggestion: _.suggestions[0] }));
+
+  potentialSuggestions.forEach(potentialSuggestion => {
+    const { condition, observation, procedure, medicationAdministration } = potentialSuggestion.suggestion;
+    const extractedCondition = {
+      resourceType: 'Condition',
+      code: resolveValue(condition.conditionCode, codingSuggestionResolver),
+      ...conditionallyAddField('onsetDateTime', resolveValue(condition.onsetDate, dateSuggestionResolver)),
+      ...conditionallyAddField('abatementDateTime', resolveValue(condition.abatementDate, dateSuggestionResolver)),
+      ...conditionallyAddField('bodySite', bodySiteResolver(condition.bodySite))
+    };
+    const extractedObservation = {
+      resourceType: 'Observation',
+      code: resolveValue(observation.observationCode, codingSuggestionResolver),
+      ...conditionallyAddField('effectiveDateTime', resolveValue(observation.date, dateSuggestionResolver)),
+      ...conditionallyAddField('valueQuantity', resolveValue(observation.value, observationValueQuantityResolver)),
+      ...conditionallyAddField('valueString', resolveValue(observation.value, observationValueStringResolver)),
+      status: 'unknown'
+    };
+    const extractedProcedure = {
+      resourceType: 'Procedure',
+      code: resolveValue(procedure.procedureCode, codingSuggestionResolver),
+      ...getDate(
+        resolveValue(procedure.date, dateSuggestionResolver),
+        resolveValue(procedure.endDate, dateSuggestionResolver),
+        'performedPeriod',
+        'performedDateTime'
+      ),
+      ...conditionallyAddField('bodySite', bodySiteResolver(procedure.bodySite)),
+      status: 'unknown'
+    };
+    const medicationDosage = resolveValue(medicationAdministration.dosage, medicationDosageResolver);
+    const extractedMedicationAdministration = {
+      resourceType: 'MedicationAdministration',
+      ...getDate(
+        resolveValue(medicationAdministration.date, dateSuggestionResolver),
+        resolveValue(medicationAdministration.endDate, dateSuggestionResolver),
+        'effectivePeriod',
+        'effectiveDateTime'
+      ),
+      medicationCodeableConcept: resolveValue(medicationAdministration.medicationCode, codingSuggestionResolver),
+      ...medicationDosage && { dosage: { text: medicationDosage } },
+      status: (medicationAdministration.status[0] && medicationAdministration.status[0].value) || 'unknown'
+    };
+    if (extractedMedicationAdministration.medicationCodeableConcept) {
+      filteredSuggestions.push(extractedMedicationAdministration);
+    }
+    if (extractedCondition.code) {
+      filteredSuggestions.push(extractedCondition);
+    }
+    if (extractedObservation.code) {
+      filteredSuggestions.push(extractedObservation);
+    }
+    if (extractedProcedure.code) {
+      filteredSuggestions.push(extractedProcedure);
+    }
+  });
```

Comment:
So if the user opts for filtered suggestions, then we only want to provide those suggestions that have a code defined for them regardless of whether other fields are defined?

Reply:
That's what it currently does, but that could be changed. Removing that check results in a lot of resources that just have a date or value field, but without a code field there is no context to what that represents

Reply:
Right. I'm wondering if the medication dosage has more meaning without the coding . . .
I think it's fine for now. We can always change later.

Reply:
That's a good point. I'll change that